integrations/oci/tls-certificates/src/main/java/io/helidon/integrations/oci/tls/certificates/LifecycleHook.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.integrations.oci.tls.certificates;
+
+import java.util.function.Consumer;
+
+import io.helidon.inject.api.Contract;
+
+// consider making this public and relocating this to somewhere under common or inject
+// it is here to ensure proper shutdown (decoupled from mp) and presenting non-intermittent test failures
+@Contract
+interface LifecycleHook {
```

Comment:
@tomas-langer I would like to discuss this with you too. This contract allows a module to tap into CDI container lifecycle without having to physically depend on CDI in the module itself.. currently it is coming in through test dependencies to guard against race conditions on shutdown - especially when the test is repeated > 1 times. While it seems to now only have test benefits, I do see a lot of value with this in the broader scheme of things. Perhaps we add this contract to the upcoming injection-cdi extension module covered by https://github.com/helidon-io/helidon/issues/6499. The contract definition would need to live in inject-api though.  But having it would be a game changer for modules to benefit from having the ability to decouple from MP/CDI.